internal/resources/cluster/resource_cluster.go

```diff
@@ -274,6 +356,66 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, m interf
 	return append(diags, dataSourceTMCClusterRead(context.WithValue(ctx, contextMethodKey{}, "create"), d, m)...)
 }
 
+type (
+	kubeConfigOption func(*kubeConfig)
+
+	kubeConfig struct {
+		filePath string
+		raw      string
+	}
+)
+
+func withPath(p string) kubeConfigOption {
+	return func(config *kubeConfig) {
+		config.filePath = p
+	}
+}
+
+func withRaw(r string) kubeConfigOption {
+	return func(config *kubeConfig) {
+		config.raw = r
+	}
+}
+
+func getK8sClient(opts ...kubeConfigOption) (*k8sClient.Client, error) {
```

Comment:
would it make sense to move this function inside the package where the k8s client is being consumed ie manifest package.

If you are moving, move both the function and the type variables

Reply:
Then we would need to export all variables and functions by making them global. Since, only cluster resource is using this, we can have it here.